testhelper/convenience.go

```diff
@@ -232,6 +232,16 @@ func CheckEquals(t *testing.T, expected, actual any) {
 	}
 }
 
+// AssertDeepEquals - like Equals - performs a comparison - but on more complex
+// structures that requires deeper inspection
+func AssertTypeEquals(t *testing.T, expected, actual any) {
+	t.Helper()
+
+	if reflect.TypeOf(expected) != reflect.TypeOf(actual) {
+		logFatal(t, fmt.Sprintf("expected %s but got %s", green(expected), yellow(actual)))
```

Comment:
don't you want to apply the same `\033[1;31m` formatting as below?

Reply:
Maybe one for a follow-up PR?

Reply:
OK